lib/api/redfish-1.0/systems.js

```diff
@@ -779,7 +780,30 @@ var getSimpleStorage = controller(function(req, res)  {
             }).catch(function(error) {
                 return redfish.handleError(error, res);
             });
-        } else {
+        } else if(result.vendor === 'Cisco'){
+            return dataFactory(identifier, 'UCS:board')
+                .then(function(catalogsData){
+                    var controllerId = 'storage-' + index.split('_')[0] + '-collection';
+                    if(catalogsData.data.children[controllerId] === undefined){
+                        throw new Errors.NotFoundError('simplestorage contorller not found');
```

Comment:
although throw can be swallowed by catch here as well, I would prefer to use Promise.reject as it consistent with the operations inside a promise train, which might be easier for reading and porting to other logic block if possible.